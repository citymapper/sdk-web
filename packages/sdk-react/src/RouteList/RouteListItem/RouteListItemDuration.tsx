import React from 'react'
import styled from '../../utils/cmStyled'
import Blip from './Blip'

interface RouteListItemDurationProps {
  minutes: number
  isLive?: boolean
  minWidth?: number
}

const Wrapper = styled('div', { className: 'cm-route-duration-root' })<{
  durationWidth?: number
}>`
  display: flex;
  align-self: flex-start;
  align-items: baseline;
  justify-content: flex-end;
  min-width: ${({ durationWidth }) => durationWidth + 'px' || ''};

  &.cm-route-duration-root {
    ${({ theme }) => theme.CmRouteDuration?.root}
  }
`

const Value = styled('div', { className: 'cm-route-duration-value' })`
  font-size: 1.125em;
  font-weight: 600;
  margin-right: ${({ theme }) => theme.spacing(0.25)};
  font-variant-numeric: tabular-nums;

  &.cm-route-duration-value {
    ${({ theme }) => theme.CmRouteDuration?.value}
  }
`

const Units = styled('div', { className: 'cm-route-duration-units' })`
  font-size: 0.6875em;
  font-weight: 400;

  &.cm-route-duration-units {
    ${({ theme }) => theme.CmRouteDuration?.units}
  }
`

const RouteListItemDuration: React.FC<RouteListItemDurationProps> = ({
  minutes,
  isLive,
  minWidth,
}) => {
  return (
    <Wrapper durationWidth={minWidth} data-is-live={isLive}>
      {isLive ? <Blip /> : null}
      <Value>{minutes}</Value>
      <Units>min</Units>
    </Wrapper>
  )
}

export default RouteListItemDuration
