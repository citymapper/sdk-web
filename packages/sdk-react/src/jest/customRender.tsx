import { render, RenderOptions } from '@testing-library/react'
import React, { JSXElementConstructor, ReactElement } from 'react'
import CMProvider from '../CMProvider'

const TestWrapper: React.FC<{ children: React.ReactElement }> = ({
  children,
}) => (
  <CMProvider
    config={{
      apiKey: '',
      citymapperFont: false,
    }}
  >
    {children}
  </CMProvider>
)

const customRender = (
  ui: ReactElement<any, string | JSXElementConstructor<any>>,
  options?: RenderOptions<
    typeof import('@testing-library/dom/types/queries'),
    HTMLElement,
    HTMLElement
  >
) => render(ui, { wrapper: TestWrapper, ...options })

export default customRender
