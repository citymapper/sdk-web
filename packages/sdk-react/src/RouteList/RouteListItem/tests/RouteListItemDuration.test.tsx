import React from 'react'
import customRender from '../../../jest/customRender'
import RouteListItemDuration from '../RouteListItemDuration'

describe('RouteListItemDuration', () => {
  it('should render just the duration', () => {
    const { container } = customRender(<RouteListItemDuration minutes={120} />)
    expect(container).toMatchSnapshot()
  })

  it('should render with min-width', () => {
    const { container } = customRender(
      <RouteListItemDuration minutes={120} minWidth={100} />
    )
    expect(container).toMatchSnapshot()
  })

  it('should render as live', () => {
    const { container } = customRender(
      <RouteListItemDuration minutes={120} isLive />
    )
    expect(container).toMatchSnapshot()
  })

  it('should render just the duration text', () => {
    const { container } = customRender(<RouteListItemDuration minutes={60} />)
    expect(container).toHaveTextContent('60min')
  })
})
