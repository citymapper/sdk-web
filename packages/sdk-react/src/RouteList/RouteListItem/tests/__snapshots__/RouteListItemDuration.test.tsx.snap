// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RouteListItemDuration should render as live 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: flex-start;
  align-self: flex-start;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  -webkit-box-pack: end;
  -ms-flex-pack: end;
  -webkit-justify-content: flex-end;
  justify-content: flex-end;
  min-width: undefinedpx;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: flex-start;
  align-self: flex-start;
  margin-right: 4px;
}

.emotion-1 svg {
  width: 8px;
  height: 8px;
}

.emotion-2 {
  width: 24px;
  height: 24px;
}

.emotion-3 {
  font-size: 1.125em;
  font-weight: 600;
  margin-right: 2px;
  font-variant-numeric: tabular-nums;
}

.emotion-4 {
  font-size: 0.6875em;
  font-weight: 400;
}

<div>
  <div
    class="cm-route-duration-root emotion-0"
    data-is-live="true"
  >
    <div
      class="cm-departure-live-blip emotion-1"
    >
      <svg
        class="emotion-2"
        fill="none"
        viewBox="0 0 8 8"
      >
        <path
          clip-rule="evenodd"
          d="M3.46447 3.46447C4.40215 2.52678 5.67392 2 7 2C7.55228 2 8 1.55228 8 1C8 0.447715 7.55228 0 7 0C5.14348 0 3.36301 0.737498 2.05025 2.05025C0.737498 3.36301 0 5.14348 0 7C0 7.55228 0.447715 8 1 8C1.55228 8 2 7.55228 2 7C2 5.67392 2.52678 4.40215 3.46447 3.46447ZM5.93934 5.93934C6.22064 5.65804 6.60218 5.5 7 5.5C7.55228 5.5 8 5.05228 8 4.5C8 3.94772 7.55228 3.5 7 3.5C6.07174 3.5 5.1815 3.86875 4.52513 4.52513C3.86875 5.1815 3.5 6.07174 3.5 7C3.5 7.55228 3.94772 8 4.5 8C5.05228 8 5.5 7.55228 5.5 7C5.5 6.60218 5.65804 6.22064 5.93934 5.93934Z"
          fill="#FFD600"
          fill-rule="evenodd"
        />
      </svg>
    </div>
    <div
      class="cm-route-duration-value emotion-3"
    >
      120
    </div>
    <div
      class="cm-route-duration-units emotion-4"
    >
      min
    </div>
  </div>
</div>
`;

exports[`RouteListItemDuration should render just the duration 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: flex-start;
  align-self: flex-start;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  -webkit-box-pack: end;
  -ms-flex-pack: end;
  -webkit-justify-content: flex-end;
  justify-content: flex-end;
  min-width: undefinedpx;
}

.emotion-1 {
  font-size: 1.125em;
  font-weight: 600;
  margin-right: 2px;
  font-variant-numeric: tabular-nums;
}

.emotion-2 {
  font-size: 0.6875em;
  font-weight: 400;
}

<div>
  <div
    class="cm-route-duration-root emotion-0"
  >
    <div
      class="cm-route-duration-value emotion-1"
    >
      120
    </div>
    <div
      class="cm-route-duration-units emotion-2"
    >
      min
    </div>
  </div>
</div>
`;

exports[`RouteListItemDuration should render with min-width 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: flex-start;
  align-self: flex-start;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  -webkit-box-pack: end;
  -ms-flex-pack: end;
  -webkit-justify-content: flex-end;
  justify-content: flex-end;
  min-width: 100px;
}

.emotion-1 {
  font-size: 1.125em;
  font-weight: 600;
  margin-right: 2px;
  font-variant-numeric: tabular-nums;
}

.emotion-2 {
  font-size: 0.6875em;
  font-weight: 400;
}

<div>
  <div
    class="cm-route-duration-root emotion-0"
  >
    <div
      class="cm-route-duration-value emotion-1"
    >
      120
    </div>
    <div
      class="cm-route-duration-units emotion-2"
    >
      min
    </div>
  </div>
</div>
`;
